website/src/pages/api/auth/[...nextauth].ts

```diff
@@ -71,6 +72,48 @@ export const authOptions: AuthOptions = {
   session: {
     strategy: "jwt",
   },
+  callbacks: {
+    async signIn(user, account) {
+      // Check if the user should be marked as an admin.
+      const adminUsers = process.env.ADMIN_USERS.split(",");
+      if (account && account.providor === "discord") {
+        const discordID = account.id;
+        const discordUsername = account.username;
+        const discordIdAndUsername = `${discordUsername}#${discordID}`;
+        if (adminUsers && adminUsers.includes(discordIdAndUsername)) {
```

Comment:
Could you turn these two checks into a function? That way we can do

const isAdminuser = checkAdminAccount(adminUsers);
if (isAdminUser) {
  // Do prisma update
}

Reply:
Also, could we only do the prisma write if the user doesn't already have the `admin` role set? Ideally we write to the database just once.